src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+        system.assertEquals('Direct City Edit', mapAccIdAcc.values()[1].MailingCity);
+        system.assertEquals(true, mapAccIdAcc.values()[1].is_Address_Override__c);
+ 
+         // verify the previous addresses didn't get Default cleared.
+        List<Address__c> ListAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Contact__c from Address__c];
+        system.assertEquals(4, ListAddr.size());
+        for (Address__c addr : ListAddr) {
+            boolean isNewEdit = (addr.MailingStreet__c.contains('Direct Street Edit'));
+            if(isNewEdit) {
+                system.assertNotEquals(false, addr.Default_Address__c);
+            }
+        }
+    }
+
+    /*********************************************************************************************************
+    @description Update Contact's Mailing address with just a case change, to cause an update to default Address object  
+    verify:
+        existing Default Address updated
+        Contact has updated Address
+    **********************************************************************************************************/            
+    @isTest
+    public static void updateAddrExistingCaseOnly() { 
+        configSettings();
+        
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+
+        // modify some of the Contact addresses directly
+        // NOTE: we only modify 1 field, so it will be treated as an update to an existing address!
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = 'Direct Street Edit';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;
+
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingState = 'Washington';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;
+
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = ' direct street  edit ';  // whitespace and casing should not count as a real edit.
+            contactsAddrs.contacts[i].MailingState = 'Oregon';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;
+
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = ' DIRECT STREET  EDIT ';  // only change casing.
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.contacts;
+        Test.stopTest();
+        
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.